### ISSUE TITLE
Create a rustfmt.toml definition

### ISSUE THREAD

**Issue by vlopes11** [0xPolygonMiden/miden-vm#456]
Consider standardizing the rustfmt definitions so we have a consistent style across the project

This will also halt the CI when there is something out of style so we automatize this review

_Originally posted by @bobbinth in https://github.com/maticnetwork/miden/pull/450#discussion_r1007247147_

**Comment by bobbinth** [comment]
If we will be defining out own rustfmt rules, maybe we could also consider setting the following parameters:
```
array_width = 80
attr_fn_like_width = 80
chain_width = 80
fn_call_width = 80
single_line_if_else_max_width = 60
```

**Comment by vlopes11** [comment]
My suggestions are

[max_width](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#max_width)
value: 110
modern IDEs are bigger than the classic 80-100 cols

[condense_wildcard_suffixes](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#condense_wildcard_suffixes)
value: true
tracking issue: #3384
we save chars on unused tuples

[enum_discrim_align_threshold](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#enum_discrim_align_threshold)
value: 40
tracking issue: #3372
enum discriminants are more readable if aligned

[fn_single_line](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#fn_single_line)
value: true
tracking issue: #3358
we save many lines of code from expressions that are of no relevance

[format_code_in_doc_comments](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#format_code_in_doc_comments)
value: true
tracking issue: #3348
everything must be formatted, including examples

[format_macro_matchers](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#format_macro_matchers)
value: true
tracking issue: #3354
ditto; everything must be formatted

[format_strings](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#format_strings)
value: true
tracking issue: #3353
We end up having to do that manually for MASM strings

[hex_literal_case](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#hex_literal_case)
value: lower
tracking issue: #5081
better to have a standard, and lowercase = less keypress

[imports_granularity](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#imports_granularity)
value: "Crate"
tracking issue: #4991
We are enforcing the crate style manually; better delegate this to rustfmt

[newline_style](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#newline_style)
value: "Unix"
Because Microsoft should never have created a new standard anyway

[normalize_doc_attributes](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#normalize_doc_attributes)
value: true
tracking issue: #3351
Kinda old style that no one uses today, but better enforce that via rustfmt so no old school Rustacean external collaborator will introduce that.

[reorder_impl_items](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#reorder_impl_items)
value: true
tracking issue: #3363
Quite important to keep this type of consistency. IMO should be a default value in rustfmt

[group_imports](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#group_imports)
value: "StdExternalCrate"
tracking issue: #5083
Easier to find dependencies if we always have the same order of imports, everywhere.

[tab_spaces](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#tab_spaces)
value: 2
I think 4 is a waste of precious columns

[use_field_init_shorthand](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#use_field_init_shorthand)
value: true
less code, same effect

[use_try_shorthand](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#use_try_shorthand)
value: true
ditto; will be useful if some old school contributes

[wrap_comments](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#wrap_comments)
value: true
tracking issue: #3347
comments should also follow format rules

**Comment by bobbinth** [comment]
Thank you! Agree with pretty much everything. The only two exceptions it that I would keep the following settings as follows:

* `max_width = 100`
* `tab_spaces = 4`

**Comment by vlopes11** [comment]
@bobbinth sounds good! The change is simple in nature but will break all ongoing PRs since it will cause merge conflicts everywhere. Should we do it now or wait for the conclusion of 0.4?

**Comment by bobbinth** [comment]
Closed by #594
